### ISSUE TITLE
��� Cannot read UNICODE environment variables from shell.

### ISSUE THREAD

**Issue by PMExtra** [docker/compose#6372]
## First: Make a `docker-compose.yml`

```yaml
version: '3.4'

services:
  test:
    image: alpine:latest
    environment:
      - ASCII_VARIABLE=${ASCII_VARIABLE}
      - CHINESE_VARIABLE=${CHINESE_VARIABLE}
      - JAPANESE_VARIABLE=${JAPANESE_VARIABLE}
```

## Second: Make the compose up by docker/compose image.

```Shell
docker run --name=compose \
  -v /var/run/docker.sock:/var/run/docker.sock \
  -v $(pwd):/test -w=/test \
  -e "ASCII_VARIABLE=test" \
  -e "CHINESE_VARIABLE=测试" \
  -e "JAPANESE_VARIABLE=テスト" \
  docker/compose:1.23.1 up -d
```

### Check:

Run `docker inspect compose`, I got the part:

```JSON
    "Env": [
        "ASCII_VARIABLE=test",
        "CHINESE_VARIABLE=测试",
        "JAPANESE_VARIABLE=テスト",
        "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
        "GLIBC=2.28-r0"
    ],
```

Run `docker inspect test`, I got the part:

```JSON
    "Env": [
        "ASCII_VARIABLE=test",
        "CHINESE_VARIABLE=������",
        "JAPANESE_VARIABLE=���������",
        "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
    ],
```

## BTW

If I create a `.env` file and pass through variables from it, then it will works all fine.

But in my case, I'm using the gitlab-ci. It pass though variables from ci settings and it will set these variables as shell env. So I really don't want a `.env` file.

Whatever, I think this is really a bug and should be fixed.

**Comment by PMExtra** [comment]
@shin- Any progress?

**Comment by HTL-chien** [comment]
Is there any progress? I notice that it seems to be fixed on `1.25.0-rc1`. When will the official `1.25.0` be released?
